### DISCUSSION TITLE
Tux should be able to wear the sunglasses (displayed in the controls icon) in game

### DISCUSSION THREAD

**Discussion by RustoMCSpit** [SuperTux/supertux#2725] [Ideas]
It'd be cool if they were a hidden unlockable or an easter egg or something

**Comment by bruhmoent** [comment]
Kindly share more detailed ideas in the "Discussions" tab instead of the issues section.

**Reply by korbanpapa** [reply]
No, this is definitely an issue. :D

**Reply by RustoMCSpit** [reply]
real and tuxpilled 
![306394710-3493f25e-47dd-4dcb-a8ad-f7867168ddca](https://github.com/user-attachments/assets/27c13f6b-60df-434b-ac18-668dee9b5709)

**Comment by tylerandari13** [comment]
To add onto this, imagine some sorta hats like the santa hat that the player could wear and the sunglasses were one of these hats. Like you said it would be really cool and make for an easier time telling players apart in multiplayer.

~~also to anyone who talks with me on the Discord i know ive mentioned this at least 3 times but its a cool idea and feels relevant to this discussion~~

**Comment by RustoMCSpit** [comment]
https://github.com/SuperTux/supertux/tree/master/data/images/creatures/tux/big

Now, we could go about this by just copying and pasting this but that isn't a very scalable approach. We'd be better using these sprites and making nodes on top of them and then making a glasses sprite however that isn't as customised

**Comment by RustoMCSpit** [comment]
If we make nodes on the sprites we can make it so that we could have an entire workshop for Tux where the community could submit their creations? Maybe they could have it so that it can change sprite on a (random?) keyframe or that it can move a certain bit off axis by a random amount if falling or whatever.

Basically:
if falling: 
     On 40 percent of the animation cycles, move y axis by random (-1, -5) on keyframe random (4, 7) and change to sprite "glasses falling".
And have the above numbers be altered by velocity

If standing:
On 5 percent of the animation cycles, change to sprite "sunglasses reflect sunlight".

you get the gist, how feasible is this for a future workshop or is this too complex? ideally we could just anchor accessories to nodes on sprites and have them controlled by basic building block code instructions and have them downloadable in game

**Comment by RustoMCSpit** [comment]
the above approach could also apply hats, glasses, cigars, monocles, and more to enemies as well. tux already has super power variants, how do those function

**Reply by RustoMCSpit** [reply]
by the looks of this https://github.com/SuperTux/supertux/tree/master/data/images/creatures/tux/santahat/big it appears we're just layering sprites on sprites. is that the most efficient approach here for scalability?
